server/cli/kv.go

```diff
@@ -0,0 +1,220 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Peter Mattis (peter.mattis@gmail.com)
+//
+// TODO(pmattis): Increment, ConditionalPut, DeleteRange.
+
+package cli
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/rpc"
+
+	commander "code.google.com/p/go-commander"
+)
+
+func makeKVClient() *client.KV {
+	tlsConfig := &http.Transport{
+		TLSClientConfig: rpc.LoadInsecureTLSConfig().Config(),
+	}
+	kv := client.NewKV(client.NewHTTPSender(Context.HTTP, tlsConfig), nil)
+	// TODO(pmattis): Initialize this to something more reasonable
+	kv.User = "root"
+	return kv
+}
+
+// A CmdGet command gets the value for the specified key.
+var CmdGet = &commander.Command{
+	UsageLine: "get [options] <key>",
+	Short:     "gets the value for a key",
+	Long: `
+Fetches and display the value for <key>.
+`,
+	Run:  runGet,
+	Flag: *flag.CommandLine,
+}
+
+func runGet(cmd *commander.Command, args []string) {
+	if len(args) != 1 {
+		cmd.Usage()
+		return
+	}
+	kv := makeKVClient()
+	defer kv.Close()
+
+	key := proto.Key(args[0])
+	resp := &proto.GetResponse{}
+	if err := kv.Call(proto.Get, proto.GetArgs(key), resp); err != nil {
+		fmt.Fprintf(os.Stderr, "get failed: %s\n", err)
+		os.Exit(1)
+	}
+	if resp.Value == nil {
+		fmt.Fprintf(os.Stderr, "%s not found\n", key)
+		os.Exit(1)
+	}
+	fmt.Printf("%s\n", resp.Value.Bytes)
+}
+
+// A CmdPut command sets the value for one or more keys.
+var CmdPut = &commander.Command{
+	UsageLine: "put [options] <key> <value> [<key2> <value2>...]",
+	Short:     "sets the value for a key",
+	Long: `
+Sets the value for one or more keys. Keys and values must be provided
+in pairs on the command line. All of the key/value pairs are set within
+a transaction.
+`,
+	Run:  runPut,
+	Flag: *flag.CommandLine,
+}
+
+func runPut(cmd *commander.Command, args []string) {
+	if len(args) == 0 || len(args)%2 == 1 {
+		cmd.Usage()
+		return
+	}
+
+	// Do not allow system keys to be put.
+	for i := 0; i < len(args); i += 2 {
+		if strings.HasPrefix(args[i], "\x00") {
```

Comment:
Is it even possible to pass `\x00` through command-line arguments? At the C level it's a `char**`; is there some other interface to get arguments that are not null-terminated?